src/DataStructures/MakeWithValue.hpp

```diff
@@ -0,0 +1,42 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+///\file
+/// Defines make_with_value
+
+#pragma once
+
+#include "Utilities/ForceInline.hpp"
+
+namespace MakeWithValueImpl_detail {
+template <typename R, typename T, typename V>
```

Comment:
move `T` and `V` onto the function, then they can be deduced, and we have fewer type instantiations.

Reply:
You can't do this, no partial specializations of functions!

Reply:
You can do it for `V` at least, no? No need to yell.

Reply:
I'm not convinced this level of generality gains us anything other than compile time...

Reply:
If you move `R` then I don't think you could ever encounter partial specialization because you cannot deduce return types. Unless there's an issue with functions differing in their return type that arises then, but I'd like to try.